pxl_scripts/px/redis_data/redis_data.pxl

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) Pixie Labs, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License")
+
+''' Redis Data Tracer
+
+This script traces all Redis data on the cluster.
+'''
+
+import px
+
+
+def redis_data(start: str, num_head: int):
+
+    # redis_events.beta is a beta table, which is subject to change.
+    df = px.DataFrame(table='redis_events.beta', start_time=start)
+    df = add_source_dest_columns(df)
+
+    # Filter source column.
+    # df = df[df.source == 'default/cartservice']
+
+    # Restrict number of results.
+    df = df.head(num_head)
+
+    df = add_source_dest_links(df, start)
+    df = df['time_', 'source', 'destination', 'remote_port', 'cmd',
+            'cmd_args', 'resp', 'latency_ns']
+
+    return df
+
+
+def add_source_dest_columns(df):
+    ''' Adds source, destination columns to the dataframe.
+
+    @df: A DataFrame containing trace_role, upid, remote_addr columns.
+
+    Returns: The input DataFrame with source, destination, pod, namespace,
+    is_source_pod_type, is_dest_pod_type columns added.
+    '''
+    df.pod = df.ctx['pod']
+    df.namespace = df.ctx['namespace']
+
+    # TODO: remove hack to get around PP-2445.
+    df = df.head(1000000000)
+
+    # If remote_addr is a pod, get its name. If not, use IP address.
+    df.ra_pod = px.pod_id_to_pod_name(px.ip_to_pod_id(df.remote_addr))
+    df.is_ra_pod = df.ra_pod != ''
+    df.ra_name = px.select(df.is_ra_pod, df.ra_pod, df.remote_addr)
+
+    df.is_server_tracing = df.trace_role == 2
+    df.is_source_pod_type = px.select(df.is_server_tracing, df.is_ra_pod, True)
+    df.is_dest_pod_type = px.select(df.is_server_tracing, True, df.is_ra_pod)
+
+    # Set source and destination based on trace_role.
+    # Redis requests are traced server-side (trace_role==2),
+    # unless the server is outside of the cluster in which
+    # case the request is traced client-side (trace_role==1).
+    df.source = px.select(df.is_server_tracing, df.ra_name, df.pod)
+    df.destination = px.select(df.is_server_tracing, df.pod, df.ra_name)
+
+    df = df.drop(['ra_pod', 'is_ra_pod', 'ra_name', 'is_server_tracing'])
+
+    return df
+
+
+def add_source_dest_links(df, start: str):
+    ''' Adds source, destination columns with script reference links to the DataFrame.
```

Comment:
same feedback here, I don't think the user understands what source, destination, script reference are so I would consider replacing the docstring you have with some high level explanation about the logic